packages/teams-js/src/private/associatedApps.ts

```diff
@@ -0,0 +1,237 @@
+          throw new Error(`Error code: ${error.errorCode}, message: ${error.message ?? 'None'}`);
+        }
+        return response as TabInstance;
+      });
+    }
+
+    /**
+     * @hidden
+     * @internal
+     * @beta
+     * Limited to Microsoft-internal use
+     *
+     * TODO: Add full description of what this function does, ie "Renames the tab associated with an app"
+     *
+     * @param tab fill in details
+     * @param threadId Info about where this comes from, links to external docs if available, etc.
+     *
+     * @returns The TabInstance of the newly renamed app
+     *
+     * @throws TODO: Description of errors that can be thrown from this function
+     */
+    export function rename(tab: TabInstance, threadId: string): Promise<TabInstance> {
+      ensureInitialized(runtime); // TODO: add frameContext checks if this is limited to certain contexts such as content
+
+      if (!isSupported()) {
+        throw new Error(ErrorCode.NOT_SUPPORTED_ON_PLATFORM.toString());
+      }
+
+      validateTab(tab);
+      validateThreadId(threadId);
+
+      return sendMessageToParentAsync<[boolean, TabInstance | ErrorResponse]>(
+        'apiVersionTag', // TODO: see uses of getApiVersionTag in other files to do this correctly
+        'associatedApps.tab.rename',
+        [tab, threadId],
+      ).then(([wasSuccessful, response]: [boolean, TabInstance | ErrorResponse]) => {
+        if (!wasSuccessful) {
+          // TODO: Can handle error codes differently here, for example if you don't want "user cancelled" to throw
+          const error = response as ErrorResponse;
+          throw new Error(`Error code: ${error.errorCode}, message: ${error.message ?? 'None'}`);
+        }
+        return response as TabInstance;
+      });
+    }
+
+    /**
+     * @hidden
+     * @internal
+     * @beta
+     * Limited to Microsoft-internal use
+     *
+     * TODO: Add full description of what this function does, ie "Removes a tab associated with an app, must be called on an app tab, etc."
+     *
+     * @param tab fill in details
+     * @param threadId Info about where this comes from, links to external docs if available, etc.
+     *
+     * @throws TODO: Description of errors that can be thrown from this function
+     */
+    export function remove(tab: TabInstance, threadId: string): Promise<void> {
+      ensureInitialized(runtime); // TODO: add frameContext checks if this is limited to certain contexts such as content
+
+      if (!isSupported()) {
+        throw new Error(ErrorCode.NOT_SUPPORTED_ON_PLATFORM.toString());
+      }
+
+      validateTab(tab);
+      validateThreadId(threadId);
+
+      return sendMessageToParentAsync<[boolean, TabInstance | ErrorResponse]>(
+        'apiVersionTag', // TODO: see uses of getApiVersionTag in other files to do this correctly
+        'associatedApps.tab.remove',
+        [tab, threadId],
+      ).then(([wasSuccessful, response]: [boolean, TabInstance | ErrorResponse]) => {
```

Comment:
Since this function is returning void, I don't think you expect to receive a `TabInstance` in the success case, right?